jolt-core/src/poly/commitment/commitment_scheme.rs

```diff
@@ -0,0 +1,97 @@
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
+
+use crate::{
+    poly::{dense_mlpoly::DensePolynomial, field::JoltField},
+    utils::{
+        errors::ProofVerifyError,
+        transcript::{AppendToTranscript, ProofTranscript},
+    },
+};
+
+#[derive(Clone, Debug)]
+pub struct GeneratorShape {
+    pub input_length: usize,
+    pub batch_type: BatchType,
+}
+
+impl GeneratorShape {
+    pub fn new(input_length: usize, batch_type: BatchType) -> Self {
+        Self {
+            input_length,
+            batch_type,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum BatchType {
+    Big,
+    Small,
+    SurgeInitFinal,
+    SurgeReadWrite,
+}
+
+pub trait CommitmentScheme: Clone + Sync + Send + 'static {
+    type Field: JoltField;
+    type Generators: Clone + Sync + Send;
+    type Commitment: Sync + Send + CanonicalSerialize + CanonicalDeserialize + AppendToTranscript;
+    type Proof: Sync + Send + CanonicalSerialize + CanonicalDeserialize;
+    type BatchedProof: Sync + Send + CanonicalSerialize + CanonicalDeserialize;
+
+    fn generators(shapes: &[GeneratorShape]) -> Self::Generators;
+    fn commit(poly: &DensePolynomial<Self::Field>, gens: &Self::Generators) -> Self::Commitment;
+    fn batch_commit(
+        evals: &[&[Self::Field]],
+        gens: &Self::Generators,
+        batch_type: BatchType,
+    ) -> Vec<Self::Commitment>;
```

Comment:
in general, a batched commitment isn't necessarily a vector of `Self::Commitment`, so we may want to have a separate associated type for `BatchedCommitment`

Reply:
Good point.

We depend on the ability to flatten the batch in a few places but notably `R1CSProof::format_commitments`. I'm not exactly sure what that means in the general case. We could have a `PCS::flatten_commitments(Vec<PCS::Commitment>) -> PCS::BatchCommitment`.

I suggest we punt on this until we need to handle this case for a specific PCS.